Northstar.Custom/mod/scripts/vscripts/gamemodes/cl_gamemode_tbag.nut

```diff
@@ -0,0 +1,73 @@
+global function ServerCallback_TeabagDenied
+
+void function ClGamemodeTbag_Init()
+{
+    // add ffa gamestate asset
+	// ClGameState_RegisterGameStateAsset( $"ui/gamestate_info_ffa.rpak" )
+
+	// add music for mode, this is copied directly from the ffa/fra music registered in cl_music.gnut
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_INTRO, "music_mp_freeagents_intro", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_INTRO, "music_mp_freeagents_intro", TEAM_MILITIA )
+
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_WIN, "music_mp_freeagents_outro_win", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_WIN, "music_mp_freeagents_outro_win", TEAM_MILITIA )
+
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_DRAW, "music_mp_freeagents_outro_lose", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_DRAW, "music_mp_freeagents_outro_lose", TEAM_MILITIA )
+
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_LOSS, "music_mp_freeagents_outro_lose", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_LOSS, "music_mp_freeagents_outro_lose", TEAM_MILITIA )
+
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_THREE_MINUTE, "music_mp_freeagents_almostdone", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_THREE_MINUTE, "music_mp_freeagents_almostdone", TEAM_MILITIA )
+
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_LAST_MINUTE, "music_mp_freeagents_lastminute", TEAM_IMC )
+	RegisterLevelMusicForTeam( eMusicPieceID.LEVEL_LAST_MINUTE, "music_mp_freeagents_lastminute", TEAM_MILITIA )
+
+	#if CLIENT
+	RegisterConCommandTriggeredCallback( "+duck", PlayerPressed_down )
+	RegisterConCommandTriggeredCallback( "+toggle_duck", PlayerPressed_downtoggle )
+	#endif
+}
+
+void function ServerCallback_TeabagConfirmed()
+{
+	// heavily based on mfd code
+	entity localPlayer = GetLocalViewPlayer()
+
+	AnnouncementData announcement = Announcement_Create( "#TBAG_TEABAG_CONFIRMED" )
+	Announcement_SetTitleColor( announcement, <1,0,0> )
+	Announcement_SetPurge( announcement, true )
+	Announcement_SetPriority( announcement, 200 ) //Be higher priority than Titanfall ready indicator etc
+	Announcement_SetSoundAlias( announcement, SFX_HUD_ANNOUNCE_QUICK )
+	Announcement_SetStyle( announcement, ANNOUNCEMENT_STYLE_QUICK )
+	AnnouncementFromClass( localPlayer, announcement )
+}
+
+void function ServerCallback_TeabagDenied()
+{
+	// heavily based on mfd code
+	entity localPlayer = GetLocalViewPlayer()
+
+	AnnouncementData announcement = Announcement_Create( "#TBAG_TEABAG_DENIED" )
+	Announcement_SetTitleColor( announcement, <1,0,0> )
+	Announcement_SetPurge( announcement, true )
+	Announcement_SetPriority( announcement, 200 ) //Be higher priority than Titanfall ready indicator etc
+	Announcement_SetSoundAlias( announcement, SFX_HUD_ANNOUNCE_QUICK )
+	Announcement_SetStyle( announcement, ANNOUNCEMENT_STYLE_QUICK )
+	AnnouncementFromClass( localPlayer, announcement )
+}
+
+#if CLIENT
+void function PlayerPressed_down( entity player )

```

Comment:
you should actually be able to this entirely on server, which would probably be easier than sending clientcommands every crouch/uncrouch, you should be able to use the functions in _codecallbacks_player_input.gnut for this

Reply:
Will take a look once I finished with irl stuff.

Reply:
Updated to use server side input callbacks. Thanks for the advice.